truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -123,9 +128,48 @@ def list_bucket_files(bucket_name, data_dir, is_trusted=False):
     return all_objects
 
 
+def parse_s3_service_account_file(file_path):
+    # open the json file
+    with open(file_path, "r") as f:
+        data = json.load(f)
+
+    # validate the data
+    if "aws_access_key_id" not in data or "aws_secret_access_key" not in data:
+        raise ValueError("Invalid AWS credentials file")
+
+    # parse the data
+    aws_access_key_id = data["aws_access_key_id"]
+    aws_secret_access_key = data["aws_secret_access_key"]
+
+    return aws_access_key_id, aws_secret_access_key
+
+
+def list_s3_bucket_files(bucket_name, data_dir, is_trusted=False):
+    if is_trusted:
+        AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY = parse_s3_service_account_file(
+            data_dir / "service_account.json"
+        )
+        session = boto3.Session(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
+        s3 = session.resource("s3")
+    else:
+        # TODO(varun): handle public s3 buckets
```

Comment:
Why todo? SHouldn't this just work for public?

Reply:
I need to test it -- we're doing the check for "s3://" or "gs://" but a public bucket would just have a url.

Reply:
Not quite yet, boto requires credentials by default.

This is the workaround and I'll add it in the next PR with the refactoring that Sid mentioned:

```
if is_trusted:
        AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY = parse_s3_service_account_file(
            data_dir / "service_account.json"
        )
        session = boto3.Session(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
        s3 = session.resource("s3")
    else:
        s3 = boto3.resource("s3", config=Config(signature_version=UNSIGNED))
```